MobiledgeX/Runtime/Scripts/MobiledgeXIntegration.cs

```diff
@@ -337,69 +305,109 @@ public async Task<String> GetURI()
+        /// <returns> boolean value </returns>
+        public bool IsEdgeEnabled()
         {
-          if (me.useOnlyWifi)
-          {
-            Debug.Log("useOnlyWifi must be false to enable edge connection");
-            return false;
-          }  
-          if (proto == GetConnectionProtocols.TCP || proto == GetConnectionProtocols.UDP)
-          {
-            if (!me.netInterface.HasCellular())
+            if (me.useOnlyWifi)
             {
-              Debug.Log(proto + " connection requires a cellular interface to run connection over edge.");
-              return false;
+#if UNITY_EDITOR
+                Debug.LogWarning("MobiledgeX: Make sure useWifiOnly is not true in production, can be used only for testing");
+                return true;
+#else
+                Debug.LogError("Device is not edge enabled. Please switch to cellular connection or use server in public cloud");
+                return false;
+                
+#endif
+            }
+            if (!me.netInterface.HasCellular() || me.netInterface.GetIPAddress(me.netInterface.GetNetworkInterfaceName().WIFI) != null)
+            {
+#if UNITY_EDITOR
+                Debug.LogError("MobiledgeX: For Testing in UnityEditor use integration.useWifiOnly(true) ,\n In Production, Edge Connection requires the cellular interface to be up and the wifi interface to be off to run connection over edge.");
+                return false;
+#else
+                Debug.LogError("MobiledgeX: Connection requires the cellular interface to be up and the wifi interface to be off to run connection over edge.");
+                return false;
+#endif
             }
-          }
-          else
-          {
-            // Connections where we cannot bind to cellular interface default to wifi if wifi is up
-            // We need to make sure wifi is off
-            if (!me.netInterface.HasCellular() || me.netInterface.HasWifi())
+
+            string cellularIPAddress = me.netInterface.GetIPAddress(me.netInterface.GetNetworkInterfaceName().CELLULAR);
+            if (cellularIPAddress == null)
             {
-              Debug.Log(proto + " connection requires the cellular interface to be up and the wifi interface to be off to run connection over edge.");
-              return false;
+                Debug.LogError("MobiledgeX: Unable to find the IP address for local cellular interface.");
+                return false;
             }
-          }
-          
-          string cellularIPAddress = me.netInterface.GetIPAddress(me.netInterface.GetNetworkInterfaceName().CELLULAR);
-          if (cellularIPAddress == null)
-          {
-            Debug.Log("Unable to find ip address for local cellular interface.");
-            return false;
-          }
-    
-          return true;
-        }
-        /// <summary>
-        /// Enum for holding GetConnection Protocols (TCP, UDP, HTTP, Websocket)
-        /// </summary>
-        public enum GetConnectionProtocols
```

Comment:
I think we still need this because TCP and UDP we bind to cellular interface, so the user could have a wifi associated ip and still be edge enabled. Don't remember what we decided though.

Reply:
I am having a problem responding on slack will send you on Google Hangouts